precompiles/avsTask/methods.go

```diff
@@ -0,0 +1,110 @@
+package task
+
+import (
+	"fmt"
+
+	exocmn "github.com/ExocoreNetwork/exocore/precompiles/common"
+	"github.com/cosmos/btcutil/bech32"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/vm"
+	cmn "github.com/evmos/evmos/v14/precompiles/common"
+	"golang.org/x/xerrors"
+)
+
+const (
+	// MethodRegisterAVSTask defines the ABI method name for the avstask
+	//  transaction.
+	MethodRegisterAVSTask      = "registerAVSTask"
+	MethodRegisterBLSPublicKey = "registerBLSPublicKey"
+	MethodGetRegisteredPubkey  = "getRegisteredPubkey"
+)
+
+// RegisterAVSTask Middleware uses exocore's default avstask template to create tasks in avstask module.
+func (p Precompile) RegisterAVSTask(
+	ctx sdk.Context,
+	_ common.Address,
+	contract *vm.Contract,
+	method *abi.Method,
+	args []interface{},
+) ([]byte, error) {
+	// check the invalidation of caller contract
+	callerAddress, _ := bech32.EncodeFromBase256("exo", contract.CallerAddress.Bytes())
+	avs, _ := p.avsKeeper.GetAVSInfo(ctx, callerAddress)
+
+	if avs.GetInfo() == nil {
```

Comment:
Please check the return of this function, should be error here indicating the avs didn't exist
`func (k Keeper) GetAVSInfo(ctx sdk.Context, addr string) (*types.QueryAVSInfoResponse, error) `

Reply:
what is `GetInfo()`? There's no such function

Reply:
This method exists